core/video/encoder.go

```diff
@@ -40,11 +40,20 @@ func init() {
 	}
 }
 
+// Checks to make sure an encoder has been set
+func GetEncoder() (string, error) {
+	if encoder == "" {
+		return encoder, fmt.Errorf("neither avconv nor ffmpeg was found")
+	} else {
+		return encoder, nil
+	}
+}
+
 // Encode will encode the frames written to the returned chan to a video that
 // can be read from the Reader.
 func Encode(ctx context.Context, settings Settings) (chan<- image.Image, io.Reader, error) {
 	if encoder == "" {
```

Comment:
Use your new function here?

```go
encoder, err := Encoder()
if err != nil {
	return nil, nil, err
}
```

Reply:
See the whole point is if I don't have an encoder and it spends 15 minutes getting the images ready and then it runs `Encode` it will be a waste of 15 minutes to find out you don't have it. The idea is to check prior. I am not sure if it would make sense to keep the check here as well

Reply:
> See the whole point is if I don't have an encoder and it spends 15 minutes getting the images ready and then it runs Encode it will be a waste of 15 minutes to find out you don't have it. 

Understood.

> The idea is to check prior. I am not sure if it would make sense to keep the check here as well

There's nothing stopping a user of the `video` package not calling your new function and just calling `Encode` as we did before. The existing code already checks for `encoder` being empty and returning an identical error message - I'm simply proposing we avoid a duplicated bit of logic and duplicated hard-coded string.

Reply:
> I am not sure if it would make sense to keep the check here as well

Fair enough - if that's the case then maybe we should have a single function that does the check, and returns an `Encoder` instance which holds the `Encode` method (or returns the error message). That way there's no duplication of error logic, and you can't forget to call the function that checks there's an encoder.